api/server/volume.go

```diff
@@ -751,35 +755,41 @@ func (vd *volAPI) enumerate(w http.ResponseWriter, r *http.Request) {
 
 	v = params[string(api.OptVolumeID)]
 	if v != nil {
-		ids = make([]string, len(v))
-		for i, s := range v {
-			ids[i] = string(s)
-		}
-	} else {
-		vls, err := volumes.EnumerateWithFilters(ctx, &api.SdkVolumeEnumerateWithFiltersRequest{
-			Name:   locator.Name,
-			Labels: locator.VolumeLabels,
-		})
-
-		//vols, err = d.Enumerate(&locator, configLabels)
-		if err != nil {
-			vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
-			return
+		vols = make([]*api.Volume, 0, len(v))
+		for _, s := range v {
+			// They asked for inspect of specific volumes. We must honor with deep inspects
+			resp, err := volumes.Inspect(ctx, &api.SdkVolumeInspectRequest{
+				VolumeId: string(s),
+				Options: &api.VolumeInspectOptions{
+					Deep: true,
+				},
+			})
+			if err == nil {
+				vols = append(vols, resp.GetVolume())
```

Comment:
can Inspect() return no error and no volumes in response (for ex. if permission is denied?)  ?

Reply:
Afaik, yes. And I think there is a lot of old code to handle that situation when calling Inspect(). For example: https://github.com/libopenstorage/stork/blob/master/drivers/volume/portworx/portworx.go#L381

Reply:
oh I mean respond.GetVolumes() may return nil, do we need to check ?

Reply:
I'll add a check for that, good point.